crates/store/src/db/models/queries.rs

```diff
@@ -1072,7 +1071,155 @@ pub(crate) fn unconsumed_network_notes(
+        diesel::dsl::sql::<diesel::sql_types::Bool>("notes.rowid >= ")
+            .bind::<diesel::sql_types::BigInt, i64>(page.token.unwrap_or_default() as i64);
+
+    // SELECT {}, rowid
+    // FROM notes
+    // LEFT JOIN note_scripts ON notes.script_root = note_scripts.script_root
+    // WHERE
+    //  execution_mode = 0 AND tag = ?1 AND
+    //  block_num <= ?2 AND
+    //  (consumed_block_num IS NULL OR consumed_block_num > ?2) AND rowid >= ?3
+    // ORDER BY rowid
+    // LIMIT ?
+    #[allow(
+        clippy::items_after_statements,
+        reason = "It's only relevant for a single call function"
+    )]
+    type RawLoadedTuple = (
+        i64,             // block_num
+        i32,             // batch_index
+        i32,             // note_index
+        Vec<u8>,         // note_id
+        i32,             // note_type
+        Vec<u8>,         // sender
+        i32,             // tag
+        i64,             // aux
+        i64,             // execution_hint
+        Vec<u8>,         // merkle_path
+        Option<Vec<u8>>, // assets
+        Option<Vec<u8>>, // inputs
+        Option<Vec<u8>>, // script
+        Option<Vec<u8>>, // serial_num
+        i64,             // rowid (from sql::<BigInt>("notes.rowid"))
+    );
+
+    #[allow(
+        clippy::items_after_statements,
+        reason = "It's only relevant for a single call function"
+    )]
+    fn split_into_raw_note_record_and_implicit_row_id(
+        tuple: RawLoadedTuple,
+    ) -> (NoteRecordRaw, i64) {
+        (
+            NoteRecordRaw {
+                block_num: tuple.0,
+                batch_index: tuple.1,
+                note_index: tuple.2,
+                note_id: tuple.3,
+                note_type: tuple.4,
+                sender: tuple.5,
+                tag: tuple.6,
+                aux: tuple.7,
+                execution_hint: tuple.8,
+                merkle_path: tuple.9,
+                assets: tuple.10,
+                inputs: tuple.11,
+                script: tuple.12,
+                serial_num: tuple.13,
+            },
+            tuple.14,
+        )
+    }
+
+    let raw = SelectDsl::select(
+        schema::notes::table.left_join(
+            schema::note_scripts::table
+                .on(schema::notes::script_root.eq(schema::note_scripts::script_root.nullable())),
+        ),
+        (
+            schema::notes::block_num,
+            schema::notes::batch_index,
+            schema::notes::note_index,
+            schema::notes::note_id,
+            // metadata
+            schema::notes::note_type,
+            schema::notes::sender,
+            schema::notes::tag,
+            schema::notes::aux,
+            schema::notes::execution_hint,
+            schema::notes::inclusion_path,
+            // details
+            schema::notes::assets,
+            schema::notes::inputs,
+            schema::note_scripts::script.nullable(),
+            schema::notes::serial_num,
+            rowid_sel.clone(),
```

Comment:
This code seems to be repeating a few times in this file - is there a way to encapsulate it somehow?

Reply:
We can define a type and should be able to use `(NotesRaw::as_select(), schema::note_scripts::script.nullable(), rowid_sel.clone())` 🤞 - `left_join` is _very_ particular what it accpets

Reply:
Also covered in https://github.com/0xMiden/miden-node/pull/1126